src/csp/csp-auto-suggest.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { Audit } from '@adobe/spacecat-shared-data-access';
+
+const AUDIT_TYPE = Audit.AUDIT_TYPES.SECURITY_CSP;
+const scriptRegExp = /<script.*?(?:\/>|<\/script>)/gs;
+const hasNonceRegExp = /<script[^>]*? nonce=/s;
+
+async function determineSuggestionsForPage(url, context, site) {
+  const { log } = context;
+
+  log.debug(`[${AUDIT_TYPE}] [Site: ${site.getId()}]: Fetching page '${url}' for CSP auto-suggest`);
+
+  const response = await fetch(url);
```

Comment:
You should probably use https://github.com/adobe/spacecat-shared/blob/main/packages/spacecat-shared-utils/src/tracing-fetch.js here to automatically add the spacecat user agent to ensure your request can bypass bot detection.